templates/rancher/overlay/opt/corral/rancher/install-rancher.sh

```diff
@@ -1,25 +1,35 @@
 #!/bin/bash
 set -ex
 
-helm repo add rancher-latest https://releases.rancher.com/server-charts/latest
+repos=("latest" "alpha" "stable")
+if [[ ! ${repos[@]} =~ ${CORRAL_rancher_chart_repo} ]]; then
+  echo 'Error: `rancher_chart_repo` must be one of ["latest", "alpha", "stable"]'
+  exit 1
+fi
+
+helm repo add "rancher-${CORRAL_rancher_chart_repo}" "https://releases.rancher.com/server-charts/${CORRAL_rancher_chart_repo}"
 helm repo update
 
 CORRAL_rancher_host=${CORRAL_rancher_host:="${CORRAL_fqdn}"}
-CORRAL_rancher_version=${CORRAL_rancher_version:=$(helm search repo rancher-latest/rancher -o json | jq -r .[0].version)}
+CORRAL_rancher_version=${CORRAL_rancher_version:=$(helm search repo rancher-"${CORRAL_rancher_chart_repo}"/rancher -o json | jq -r .[0].version)}
 
 args=("--install" "--create-namespace")
 
 if [ -z "${CORRAL_registry_fqdn}" ]; then
-	args+=("--set hostname=${CORRAL_rancher_host}" "--version ${CORRAL_rancher_version}")
-	if [ "${CORRAL_rancher_image_tag}" ]; then
-		args+=("--set rancherImageTag=${CORRAL_rancher_image_tag}")
-	fi
+  args+=("--set hostname=${CORRAL_rancher_host}" "--version ${CORRAL_rancher_version}")
+  if [ "${CORRAL_rancher_image_tag}" ]; then
+    args+=("--set rancherImageTag=${CORRAL_rancher_image_tag}")
+  fi
 else
-	args+=("--set hostname=${CORRAL_rancher_host}" "--set rancherImage=${CORRAL_registry_fqdn}/rancher/rancher" "--set systemDefaultRegistry=${CORRAL_registry_fqdn}" "--version ${CORRAL_rancher_version}")
+  args+=("--set hostname=${CORRAL_rancher_host}" "--set rancherImage=${CORRAL_registry_fqdn}/rancher/rancher" "--set systemDefaultRegistry=${CORRAL_registry_fqdn}" "--version ${CORRAL_rancher_version}")
+fi
+
+if [ ! -z "${CORRAL_lets_encrypt_email}" ]; then
+  args+=("--set ingress.tls.source=letsEncrypt" "--set letsEncrypt.email=${CORRAL_lets_encrypt_email}" "--set letsEncrypt.ingress.class=nginx")
 fi
 
-args+=("--devel" "--wait" "-n cattle-system" "rancher rancher-latest/rancher")
-eval "helm upgrade ${args[@]}"
+args+=("--devel" "--wait" "-n cattle-system" "rancher rancher-${CORRAL_rancher_chart_repo}/rancher")
+helm upgrade "${args[@]}"
```

Comment:
I don't remember the reason of adding `eval` but I think it had to do with some parameters using certain characters.
As long as this work with the different cert types I think it should be good.

Reply:
Bash will expand the strings in `"${args[@]}"` so this should be a safer implementation.
See https://www.shellcheck.net/wiki/SC2294 and https://www.shellcheck.net/wiki/SC2145

Reply:
Yeah eval is not safe when dealing with interactive input but that worked. I'm just making sure if it worked it will be way better without it.